frontend/dist/assets/tracim/appInterface.js

```diff
@@ -21,92 +28,143 @@
+  } else {
+    selectedApp.renderAppFullscreen(app)
+    selectedApp.isRendered = true
+    ;(getSelectedApp(prevSelectedAppFullScreen) || {isRendered: null}).isRendered = false
+    prevSelectedAppFullScreen = selectedApp.name
   }
+}
+
+function GLOBAL_renderAppPopupCreation (app, retryCount) {
+  console.log('%cGLOBAL_renderAppPopupCreation', 'color: #5cebeb', app)
+
+  var selectedApp = getSelectedApp(app.config.slug)
+
+  if (selectedApp === APP_NOT_LOADED) {
+    retryCount = retryCount || 0 // INFO - CH - 2019-06-18 - old school way for default param
+
+    console.log(app.config.slug + ' does not exists. Maybe it hasn\'t finished loading yet ? Retrying in ' + TIME_TO_RETRY + 'ms')
+    var retryTime = retryCount + TIME_TO_RETRY
 
-  GLOBAL_eventReducer = ({detail: {type, data}}) => {
-    switch (type) {
-      case 'hide_popupCreateContent':
-      case 'hide_popupCreateWorkspace':
-        console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
-        getSelectedApp(data.name).unmountApp('popupCreateContentContainer')
-        break
-      case 'unmount_app':
-        console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
-
-        if (prevSelectedAppFeature !== '') {
-          const selectedApp = getSelectedApp(prevSelectedAppFeature)
-          selectedApp.unmountApp('appFeatureContainer')
-          selectedApp.unmountApp('popupCreateContentContainer')
-          selectedApp.isRendered = false
-          prevSelectedAppFeature = ''
-        }
-
-        if (prevSelectedAppFullScreen !== '') {
-          const selectedApp = getSelectedApp(prevSelectedAppFullScreen)
-          selectedApp.unmountApp('appFullscreenContainer')
-          selectedApp.isRendered = false
-          prevSelectedAppFullScreen = ''
-        }
-        break
+    if (retryTime < RETRY_TIMEOUT) {
+      setTimeout(function () {GLOBAL_renderAppPopupCreation(app, retryTime)}, TIME_TO_RETRY)
+      return
     }
+
+    console.error('Timed out waiting for app ' + app.config.slug + ' to exists')
+    return
+  }
+
+  getSelectedApp(app.config.slug).renderAppPopupCreation(app)
+}
+
+function GLOBAL_dispatchEvent (event) {
+  var type = event.type
+  var data = event.data
+  console.log('%cGLOBAL_dispatchEvent', 'color: #fff', type, data)
+
+  var customEvent = new CustomEvent('appCustomEvent', {detail: {type, data}})
+  document.dispatchEvent(customEvent)
+}
+
+function GLOBAL_eventReducer (event) {
+  var type = event.detail.type
+  var data = event.detail.data
+
+  switch (type) {
+    case 'hide_popupCreateContent':
+    case 'hide_popupCreateWorkspace':
+      console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
+      getSelectedApp(data.name).unmountApp('popupCreateContentContainer')
+      break
+    case 'unmount_app':
+      console.log('%cGLOBAL_eventReducer Custom Event', 'color: #28a745', type, data)
+
+      var selectedApp
+
+      if (prevSelectedAppFeature !== '') {
```

Comment:
what is `prevSelectedAppFeature`? A global var? What does `prev` stand for? previous?

Reply:
"previously"
I'll rename to remove abbreviation